src/Data/Text/Builder/Linear/Char.hs

```diff
@@ -74,3 +90,112 @@ unsafePrependCharM marr off c = case utf8Length c of
     A.unsafeWrite marr (off - 2) n2
     A.unsafeWrite marr (off - 1) n3
     pure 4
+
+--------------------------------------------------------------------------------
+-- Multiple chars
+--------------------------------------------------------------------------------
+
+-- | Prepend a given count of a 'Char' to a 'Buffer'.
+--
+-- >>> :set -XLinearTypes
+-- >>> runBuffer (\b -> prependChars 3 'x' (b |>. 'A'))
+-- "xxxA"
+prependChars ∷ Word → Char → Buffer ⊸ Buffer
+prependChars count ch buff
+  | count == 0 = buff
+  | isAscii ch =
+      prependExact
+        (fromIntegral count)
+        (\dst dstOff → unsafeReplicate dst dstOff (fromIntegral count) (ord ch))
+        buff
+  | otherwise =
+      case utf8Length ch of
+        cLen → case cLen * fromIntegral count of
+          totalLen →
+            prependExact
+              totalLen
+              (\dst dstOff → unsafeWrite dst dstOff ch *> unsafeTile dst dstOff totalLen cLen)
+              buff
+
+-- | Apppend a given count of a 'Char' to a 'Buffer'.
+--
+-- >>> :set -XLinearTypes
+-- >>> runBuffer (\b -> appendChars 3 'x' (b |>. 'A'))
+-- "Axxx"
+appendChars ∷ Word → Char → Buffer ⊸ Buffer
+appendChars count ch buff
+  | count == 0 = buff
+  | isAscii ch =
+      appendExact
+        (fromIntegral count)
+        (\dst dstOff → unsafeReplicate dst dstOff (fromIntegral count) (ord ch))
+        buff
+  | otherwise =
+      case utf8Length ch of
+        cLen → case cLen * fromIntegral count of
+          totalLen →
+            appendExact
+              totalLen
+              (\dst dstOff → unsafeWrite dst dstOff ch *> unsafeTile dst dstOff totalLen cLen)
+              buff
+
+--------------------------------------------------------------------------------
+-- Padding
+--------------------------------------------------------------------------------
+
+-- | Pad a builder from the /left/ side to the specified length with the specified
+-- character.
+--
+-- >>> :set -XLinearTypes
+-- >>> runBuffer (\b -> justifyRight 10 'x' (appendChars 3 'A' b))
+-- "xxxxxxxAAA"
+-- >>> runBuffer (\b -> justifyRight 5 'x' (appendChars 6 'A' b))
+-- "AAAAAA"
```

Comment:
Please add a usage example with `newEmptyBuffer`, it's untrivial to come up with for a new user.

Reply:
Done